sdk/core/Azure.Core/src/Response.cs

```diff
@@ -41,6 +41,39 @@ public abstract class Response : IDisposable
         /// </summary>
         public virtual ResponseHeaders Headers => new ResponseHeaders(this);
 
+        /// <summary>
+        /// Gets the contents of HTTP response, if it is available.
+        /// </summary>
+        /// <remarks>
+        /// Throws <see cref="InvalidOperationException"/> when <see cref="ContentStream"/> is not a <see cref="MemoryStream"/>.
+        /// </remarks>
+        public virtual BinaryData Content
+        {
+            get
+            {
+                if (ContentStream == null)
+                {
+                    return new BinaryData(Array.Empty<byte>());
+                }
+
+                MemoryStream? memoryContent = ContentStream as MemoryStream;
+
+                if (memoryContent == null)
+                {
+                    throw new InvalidOperationException($"The {nameof(ContentStream)} is not a ${typeof(MemoryStream)}");
```

Comment:
I think we should change the message. The user won't know why it's not memory stream. Maybe we should just say that this response contains non-seekable stream.

Reply:
I will make the change, but I think no matter what if the user hits this exception they are going to be confused as to why it didn't work.  It's not any clearer as to why the stream isn't seakable vs why it isn't a memory stream and the stream being seekable isn't actually sufficient for this property to return a BinaryData.  The stream *must* be of type `MemoryStream` and we *must* be able to access it's buffer.